components/kyma-operator/pkg/kymaoperation/steps/creator.go

```diff
@@ -0,0 +1,107 @@
+package steps
+
+import (
+	"log"
+
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/apis/installer/v1alpha1"
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/config"
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/kymahelm"
+	"github.com/kyma-project/kyma/components/kyma-operator/pkg/overrides"
+	errors "github.com/pkg/errors"
+	helm "k8s.io/helm/pkg/proto/hapi/release"
+)
+
+// stepFactoryCreator is used to create StepFactory instances for installation or uninstallation.
+type stepFactoryCreator struct {
+	helmClient          kymahelm.ClientInterface
+	sourceGetterSupport SourceGetterLegacySupport
+}
+
+// NewStepFactoryCreator returns a new stepFactoryCreator instance.
+func NewStepFactoryCreator(helmClient kymahelm.ClientInterface, sgls SourceGetterLegacySupport) *stepFactoryCreator {
+	return &stepFactoryCreator{
+		helmClient:          helmClient,
+		sourceGetterSupport: sgls,
+	}
+}
+
+func (sfc *stepFactoryCreator) getInstalledReleases() (map[string]kymahelm.ReleaseStatus, error) {
+
+	existingReleases := make(map[string]kymahelm.ReleaseStatus)
+
+	list, err := sfc.helmClient.ListReleases()
+	if err != nil {
+		return nil, errors.New("Helm error: " + err.Error())
+	}
+
+	if list != nil {
+		log.Println("Helm releases list:")
+
+		for _, release := range list.Releases {
+			var lastDeployedRev int32
+
+			statusCode := release.Info.Status.Code
```

Comment:
If we don't do anything with `release.Info.Status.Code`, maybe just use it directly?

Reply:
Done.